crates/rustc_codegen_spirv/src/symbols.rs

```diff
@@ -493,6 +529,136 @@ pub fn parse_attrs(
     result.collect::<Vec<_>>().into_iter()
 }
 
+fn parse_image(cx: &CodegenCx<'_>, attr: &NestedMetaItem) -> Option<SpirvAttribute> {
+    let args = match attr.meta_item_list() {
+        Some(args) => args,
+        None => {
+            cx.tcx
+                .sess
+                .span_err(attr.span(), "image attribute must have arguments");
+            return None;
+        }
+    };
+    if args.len() != 6 && args.len() != 7 {
```

Comment:
I think we can set a lot of defaults values. Having to set depth to 0 for 90% of textures seems a bit annoying.

Reply:
I don't imagine this attribute being used out of `spirv-std` (hence me leaving it undocumented), so user convenience and defaulting and whatnot isn't necessary.

Reply:
Oh okay nvm then :)

Reply:
Could we specify some of these with `const` generics instead of attributes?